### PR TITLE
Include owner as an AMI filter

### LINKED ISSUE #486: add ami owner to filter list
/kind bug

to prevent hijacking of ami name

### PR DESCRIPTION
**Pull request by chuckha**
Signed-off-by: Chuck Ha <chuckh@vmware.com>

**What this PR does / why we need it**:
this pr adds an owner-id to the ami filter.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #486 

**Special notes for your reviewer**:
Additional cleanups can be added in a later commit

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/cloud/aws/services/ec2/ami.go [hunk 1]
```diff
@@ -27,17 +27,27 @@ import (
 	"github.com/pkg/errors"
 )
 
-func amiName(platform, platformVersion, kubernetesVersion string) string {
-	return fmt.Sprintf("ami-%s-%s-%s-00-??????????", platform, platformVersion, strings.TrimPrefix(kubernetesVersion, "v"))
+const (
+	// This is a heptio owned account. Please see:
```
Comment (vincepri) [inline pkg/cloud/aws/services/ec2/ami.go]: `s/heptio/heptio (VMware)`

File: pkg/cloud/aws/services/ec2/ami.go [hunk 2]
```diff
@@ -27,17 +27,27 @@ import (
...
+	// https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/487
+	ownerID = "258751437250"
```
Comment (vincepri) [inline pkg/cloud/aws/services/ec2/ami.go]: Can we prefix this const given the package isn't just for AMIs?

File: pkg/cloud/aws/services/ec2/ami.go [hunk 3]
```diff
@@ -27,17 +27,27 @@ import (
...
+)
+
+func amiName(baseOS, baseOSVersion, kubernetesVersion string) string {
```
Comment (vincepri) [inline pkg/cloud/aws/services/ec2/ami.go]: Can we add a description on how the image name is structured?

**Comment by chuckha** [comment]
@vincepri changes made, thanks for the feedback

**Pull request merged**